src/bot/integrations/discord.ts

```diff
@@ -373,6 +410,67 @@ class Discord extends Integration {
     }
   }
 
+  @onChange('onlinePresenceStatusOnStreamName')
+  @onChange('onlinePresenceStatusDefaultName')
+  @onChange('onlinePresenceStatusOnStream')
+  @onChange('onlinePresenceStatusDefault')
+  async changeClientOnlinePresence() {
```

Comment:
you can add @onStreamStart() and @onStreamEnd() here as well next time, but that's matter of preference